File: app/src/main/java/de/schooltec/datapass/datasupplier/DummyDataSupplier.java
package de.schooltec.datapass.datasupplier;

import android.content.Context;

/**
 * DummyDataSupplier which simply says that it can't retrieve any data.
 *
 * @author Martin Hellwig
 */
class DummyDataSupplier extends DataSupplier
{
    @Override
    public ReturnCode getData(Context context)
    {
        return ReturnCode.CARRIER_UNAVAILABLE;
    }

    @Override
    public boolean isRealDataSupplier()
    {
        return false;
    }

    @Override
    public String getTrafficWasted()
    {
        return "";
    }

    @Override
    public String getTrafficAvailable()
    {
        return "";
    }

    @Override
    public String getTrafficUnit()
    {
        return "";
    }

    @Override
    public int getTrafficWastedPercentage()
    {
        return 0;
    }

    @Override
    public String getLastUpdate()
    {
        return "";
    }
}

File: app/src/main/java/de/schooltec/datapass/datasupplier/DummyDataSupplier.kt
package de.schooltec.datapass.datasupplier

import android.content.Context
import java.util.*

/**
 * DummyDataSupplier which simply says that it can't retrieve any data.
 *
 * @author Martin Hellwig
 */
internal class DummyDataSupplier : DataSupplier {
    override val isRealDataSupplier: Boolean
        get() = false

    override val trafficWasted: Long
        get() = 0

    override val trafficAvailable: Long
        get() = 0

    override val lastUpdate: Date
        get() = Date()

    override fun fetchData(context: Context): DataSupplier.ReturnCode {
        return DataSupplier.ReturnCode.CARRIER_UNAVAILABLE
    }
}
